client/store/store.ts

```diff
@@ -8,6 +8,9 @@ import { immer } from 'zustand/middleware/immer'
 
 enableMapSet()
 
+// TODO: Remove when a better solution that works for metadata editors are found
+setAutoFreeze(false)
```

Comment:
what is this doing exactly? why is it needed?

Reply:
Yea, that's a problematic point. We use `zustand` with `immer` to make store manipulations easier regarding DX but `immer` freezes all the objects that go through the store meaning that when we pass e.g. `resource` object to the Resource Editor it can't be updated. 

This line solves this problem but of course, it would be great to find a better way as freezing is quite useful to ensure correct store usage

Reply:
Even when it is too large of a comment I would move your answer @roll to the code.

This lines of code are specially difficult to maintain without proper context so I would add a little bit of verbosity on why it is needed :)

That would help future maintainers understand why is it there. I would suggest something like:

```
// We use zustand with immer to make store manipulations easier but immer freezes all the objects that go 
// through the store. This means that when we pass e.g. resource object to the Resource Editor it can't be 
// updated.
setAutoFreeze(false)
```

Reply:
I agree